src/sleepwellbaby/utils.py

```diff
@@ -76,13 +81,24 @@ def get_swb_predictions(
         if row[ref_columns].isna().any():
             pred = 'ineligible'
             proba_dict = {'AS': -1, 'QS': -1, 'W': -1}
+            X = None
         else:
             payload = convert_to_payload(temp, birth_date=birth_date, gestation_period=gestation_period)
-            pred, proba_dict = get_prediction(payload, model, model_support_dict)
+
+            if return_features:
+                pred, proba_dict, X = get_prediction(payload, model, model_support_dict, return_features)
+            else:
+                pred, proba_dict = get_prediction(payload, model, model_support_dict, return_features)
```

Comment:
This is the same function twice right? I would just run the function once and save the result to a variable and then depending on the length of the variable split in two or three (or in later Python versions use pattern matching)

```python
pred_tuple = get_prediction(payload, model, model_support_dict, return_features)
if return_features:
    pred, proba_dict, X = pred_tuple
else:
    pred, proba_dict = pred_tuple
```